packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx

```diff
@@ -503,15 +506,113 @@ const PublishAction: DocumentActionComponent = ({
     ({ canPublish, canCreate, canUpdate }) => ({ canPublish, canCreate, canUpdate })
   );
   const { publish } = useDocumentActions();
+  const [
+    countDraftRelations,
+    { isLoading: isLoadingDraftRelations, isError: isErrorDraftRelations },
+  ] = useGetDraftRelationCountQuery();
+  const [countOfDraftRelations, setCountOfDraftRelations] = React.useState(0);
+
   const [{ query, rawQuery }] = useQueryParams();
   const params = React.useMemo(() => buildValidParams(query), [query]);
+
   const modified = useForm('PublishAction', ({ modified }) => modified);
   const setSubmitting = useForm('PublishAction', ({ setSubmitting }) => setSubmitting);
   const isSubmitting = useForm('PublishAction', ({ isSubmitting }) => isSubmitting);
   const validate = useForm('PublishAction', (state) => state.validate);
   const setErrors = useForm('PublishAction', (state) => state.setErrors);
   const formValues = useForm('PublishAction', ({ values }) => values);
 
+  React.useEffect(() => {
+    if (isErrorDraftRelations) {
+      toggleNotification({
+        type: 'danger',
+        message: formatMessage({
+          id: getTranslation('error.records.fetch-draft-relatons'),
+          defaultMessage: 'An error occurred while fetching draft relations on this document.',
+        }),
+      });
+    }
+  }, [isErrorDraftRelations, toggleNotification, formatMessage]);
+
+  React.useEffect(() => {
+    let totalDraftRelations = 0;
+    const localDraftRelations = new Set();
```

Comment:
@Marc-Roig 👋🏻 this PR is FE stuff but this is the part I'd like to get your opinion on as it relates to the relations connecting syntax.

It's used to count any draft relations that are only stored in the FE edit view form based on 'connect' relation arrays

Reply:
>FE edit view form based on 'connect' relation arrays

should this take into account the FE local relations + the ones stored in database?